src/pie_utils/processor/document/split_into_partitions.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import json
+import logging
+import re
+from typing import Any, Callable, Iterable, Iterator, Match
+
+from pytorch_ie.annotations import Span
+
+from pie_utils.span.slice import have_overlap
+from pie_utils.statistics import WithStatistics
+
+from ..document import DocumentWithPartition
+
+logger = logging.getLogger(__name__)
+
+
+def get_partitions_with_matcher(
+    document: DocumentWithPartition,
+    matcher: Callable[[str], Iterable[Match]],
+    label_group_id: int | None = None,  # = 1,
+    label_whitelist: list[str] | None = None,
+    skip_initial_partition: bool = False,  # = True
+) -> Iterator[Span]:
+    """Spans are created starting with the beginning of matching entries end ending with the start
+    of the next matching one or the end of the document.
+
+    Entries with xml node names that are not in label_whitelist, if it is set, will not be
+    considered as match, i.e. their content will be added to the previous matching entry. If
+    label_group_id is set, the content of the respective match group will be taken as label.
```

Comment:
this is not the case for the current implementation, we create `Span`s without label. However, since this information may be useful, maybe we switch to `LabeledSpan` for partition? What do you think?

Reply:
Yeah as we discussed, labels might be useful in doing filtered evaluation on specific part of the document.